gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcClientHeaderProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017, Google LLC All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.rpc.ApiClientHeaderProvider;
+
+/**
+ * Transport-specific implementation of HeaderProvider that provides headers describing the API
+ * client library making API calls.
+ */
+@BetaApi("The surface for customizing headers is not stable yet and may change in the future.")
+public class GrpcClientHeaderProvider extends ApiClientHeaderProvider {
+  private static final long serialVersionUID = -2345236249769983493L;
+
+  protected GrpcClientHeaderProvider(Builder builder) {
+    super(builder);
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder extends ApiClientHeaderProvider.Builder {
+    public Builder() {
+      super();
+      setTransportToken("grpc", GaxGrpcProperties.getGrpcVersion());
```

Comment:
I'd prefer to avoid inheritance for a case like this.

Reply:
I would prefer too. I changed `GrpcClientHeaderProvider` to not have its own builder. This would also mean, that it will never be instantiated. I would still like to keep it extending `ApiClientHeaderProvider` as it seems more consistent and better suited for future modifications.

Reply:
I personally don't want future modifications along these lines either. I might prefer this class to have a single static method like the following:

```
public static ApiClientHeaderProvider withTransportToken(ApiClientHeaderProvider provider) {
  // here: convert provider to builder, set transport token using "grpc", re-build()
}
```